aws/ecs_cluster_test.go

```diff
@@ -0,0 +1,130 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test that we can succesfully list ECS clusters by manually creating a cluster and then using the list function to find it.
+func TestCanCreateAndListEcsCluster(t *testing.T) {
+	t.Parallel()
+
+	region := "eu-west-1"
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	clusterName := "test-ina-1"
+	cluster := createEcsFargateCluster(t, awsSession, clusterName)
+	defer deleteEcsCluster(awsSession, cluster)
+
+	clusterArns, err := getAllEcsClusters(awsSession)
+	if err != nil {
+		assert.Failf(t, "Unable to fetch clusters: %s", err.Error())
+	}
+
+	assert.Contains(t, clusterArns, cluster.ClusterArn)
+}
+
+// Test we can create a cluster, tag it, and then find the cluster
+func TestCanTagEcsClusterAndFilterByTag(t *testing.T) {
+	t.Parallel()
+
+	region := "eu-west-1"
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	// create a cluster without tag
+	cluster1 := createEcsFargateCluster(t, awsSession, "test-ina-3-1")
+	cluster2 := createEcsFargateCluster(t, awsSession, "test-ina-3-2")
+	defer deleteEcsCluster(awsSession, cluster1)
+	defer deleteEcsCluster(awsSession, cluster2)
```

Comment:
Put each `defer` immediately after the corresponding `createXXX`. Otherwise, if, for example, the first `create` worked, but the second failed, the code will never get to the `defer` methods for cleanup.